assembly/src/assembler/instruction/ext2_ops.rs

```diff
@@ -0,0 +1,149 @@
+use super::{AssemblyError, CodeBlock, Operation::*, SpanBuilder};
+use vm_core::{AdviceInjector::Ext2Inv, Decorator, Felt, Operation};
+
+/// Given a stack in the following initial configuration [a1, a0, b1, b0, ...] where a = (a0, a1)
+/// and b = (b0, b1) represent elements in the extension field of degree 2, this series of
+/// operations outputs the result c = (c1, c0) where c1 = a1 + b1 and c0 = a0 + b0.
+pub fn ext2_add(span: &mut SpanBuilder) -> Result<Option<CodeBlock>, AssemblyError> {
+    #[rustfmt::skip]
+    let ops = [
+        Swap,           // [a0,a1,b1,b0,...]
+        MovUp3,         // [b0,a0,a1,b1,...]
+        Add,            // [b0+a0,a1,b1,...]
+        MovDn2,         // [a1,b1,b0+a0,...]
+        Add             // [a1+b1,b0+a0,...]
```

Comment:
minor nit: would be great to add spaces after commas in comments here and other places.